src/utils.test.js

```diff
@@ -573,6 +578,175 @@ describe('isFormEditing', () => {
   });
 });
 
+describe('getNextSelectedRowsState', () => {
+  it('should add a row to selectedRows if not selected', () => {
+    const selectedRows = {};
+    const row = { id: 1, name: 'John Doe' };
+
+    const result = getNextSelectedRowsState(selectedRows, row);
+
+    expect(result).toEqual({ [row.id]: row });
+  });
+
+  it('should remove a row from selectedRows if already selected', () => {
+    const row = { id: 1, name: 'John Doe' };
+    const selectedRows = { [row.id]: row };
+
+    const result = getNextSelectedRowsState(selectedRows, row);
+
+    expect(result).toEqual({});
+  });
+
+  it('should not mutate the original selectedRows object', () => {
+    const row = { id: 1, name: 'John Doe' };
+    const selectedRows = { [row.id]: row };
+
+    const result = getNextSelectedRowsState(selectedRows, row);
+
+    expect(selectedRows).toEqual({ [row.id]: row });
+    expect(result).not.toBe(selectedRows);
+  });
+});
+describe('isPrintable', () => {
+  it('should return true when pickSlips contain a match for requestId', () => {
+    const requestId = '123';
+    const pickSlips = [
+      { request: { requestID: '123' } },
+      { request: { requestID: '456' } },
+    ];
+
+    const result = isPrintable(requestId, pickSlips);
+
+    expect(result).toBe(true);
+  });
+
+  it('should return false when pickSlips do not contain a match for requestId', () => {
+    const requestId = '789';
+    const pickSlips = [
```

Comment:
Could you create a constant in this file for `pickSlips`?
I see that you use the same object at least two times.